src/game.cpp

```diff
@@ -4106,62 +4106,162 @@ void game::debug()
     case 14: {
         point pos = look_around();
         int npcdex = npc_at(pos.x, pos.y);
-        if (npcdex == -1) {
-            popup(_("No NPC there."));
+        if( npcdex == -1 && pos != u.pos() ) {
+            popup(_("No character there."));
         } else {
-            std::stringstream data;
-            npc *p = active_npc[npcdex];
+            player &p = npcdex != -1 ? *active_npc[npcdex] : u;
+            npc *np = active_npc[npcdex];
```

Comment:
I see you need this inside the `if( p.is_npc() )` branch below. You should declare it there, because here `npcdex` can be -1, which can make this undefined behavior.